zerver/tests/test_message_move_topic.py

```diff
@@ -45,6 +46,34 @@ def check_topic(self, msg_id: int, topic_name: str) -> None:
         msg = Message.objects.get(id=msg_id)
         self.assertEqual(msg.topic_name(), topic_name)
 
+    def assert_move_message(
+        self,
+        user: str,
+        orig_stream: Stream,
+        stream_id: int | None = None,
+        topic_name: str | None = None,
+        error_msg: str | None = None,
+    ) -> None:
+        user_profile = self.example_user(user)
+        self.login(user)
```

Comment:
Use `api_patch` rather than `login`; one basically never wants a test helper changing which user is acting invisibly through calling the helper like this does.